tests/performance/event_generator/src/event_generator/event_generator.py

```diff
@@ -0,0 +1,240 @@
+"""Event generator library.
+
+This library includes classes for simulating log and file operation events, designed to test and
+validate the functionality of monitoring systems. It enables the controlled generation of events
+such as file modifications, creations, and deletions, as well as structured log output.
+
+Classes:
+    - EventGenerator: The abstract base class for all event generators. Requires subclasses
+      to implement the `generate_event` method.
+    - LogEventGenerator: Generates simulated log entries to a file, managing log rotation
+      based on size constraints.
+    - SyscheckEventGenerator: Simulates file system changes (create, modify, delete)
+      and manages state across events.
+
+The library provides a configurable environment for testing system responses to a variety
+of file and log-related events.
+"""
+
+
+import json
+import logging
+import os
+import random
+import threading
+import time
+from datetime import datetime
+
+
+class EventGenerator:
+    """Base class for generating simulation events.
+
+    This class provides the foundational attributes and methods for generating events.
+    Subclasses implement the `generate_event` method to specify the actual event
+    generation logic based on the specific type of event being simulated.
+    """
+    def __init__(self, rate: int, path: str, operations: int):
+        """Initialize the EventGenerator.
+
+        Args:
+            rate (int): The rate at which events are generated per second.
+            path (str): The filesystem path where events will be stored or handled.
+            operations (int): The total number of events to generate before stopping.
+        """
+        self.rate = rate  # events per second
+        self.path = path
+        self.operations = operations
+        self.stop_event = threading.Event()
+        self.count = 0  # counter for events
+
+    def generate_event(self) -> None:
+        """Generate an event. This method should be overridden by subclasses to produce specific types of events.
+
+        Raises:
+            NotImplementedError: If the subclass does not override this method.
+        """
+        raise NotImplementedError(
+            "This method should be overridden by subclasses.")
+
+    def start(self) -> None:
+        """Begin generating events until the stop condition or operation count is met."""
+        next_time = time.time() + 1 / self.rate
+        while not self.stop_event.is_set() and self.count < self.operations:
+            self.generate_event()
+            self.count += 1
+            next_time += 1 / self.rate
+            sleep_time = next_time - time.time()
+            if sleep_time > 0:
+                time.sleep(sleep_time)
+
+    def stop(self) -> None:
+        """Stop the event generation by setting the stop event."""
+        self.stop_event.set()
+
+
+class LogEventGenerator(EventGenerator):
```

Comment:
I will make all methods private except run and stop

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/70cf83bd07d20a1157bb1cef60c551d24b14f51e